api/v1beta1/azurecluster_webhook.go

```diff
@@ -47,6 +48,14 @@ func (c *AzureCluster) Default() {
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (c *AzureCluster) ValidateCreate() error {
+	// NOTE: AzureCluster with ExtendedLocation is behind EdgeZone feature gate flag; the web hook
+	// must prevent creating new objects in case the feature flag is disabled.
+	if !feature.Gates.Enabled(feature.EdgeZone) && c.Spec.ExtendedLocation != nil {
```

Comment:
why is this not part of the `validateCluster` function?

Reply:
It has been moved into `validateCluster` function.